syft/pkg/cataloger/php/parse_composer_lock.go

```diff
@@ -0,0 +1,50 @@
+package php
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+
+	"github.com/anchore/syft/syft/pkg"
+	"github.com/anchore/syft/syft/pkg/cataloger/common"
+)
+
+type ComposerLock struct {
+	Packages   []Dependency `json:"packages"`
+	PackageDev []Dependency `json:"packages-dev"`
+}
+
+type Dependency struct {
```

Comment:
in the future we can extract more information from the underlying JSON structure and add a new `pkg.PhpComposerMetadata` type and struct to capture additional data. 

For this PR getting composer support is a good enough step to get in on it's own 👍 .